src/lx/out/c.c

```diff
@@ -526,7 +527,7 @@ out_io(FILE *f)
 		fprintf(f, "\n");
 		fprintf(f, "\tif (c == '\\n') {\n");
 		fprintf(f, "\t\tlx->end.line--;\n");
-		fprintf(f, "\t\tlx->end.col = 0; /* XXX: lost information */\n");
+		fprintf(f, "\t\tlx->end.col = lx->end.saved_col;\n");
```

Comment:
That's a surprisingly simple solution. It hadn't occurred to me at all!

Reply:
I'm not sure it's right, it seems to give the wrong result when the input string has newlines, like `x\ny\nz\n`.